### PR TITLE
feat(nuxt): add payload control to `NuxtIsland` with `useCache` and `setCache`

### PR DESCRIPTION
**Pull request by huang-julien**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Hi :wave: 
This PR gives to the user a way to control `NuxtIsland`'s behavior at a granular level since it uses the app context to share the cache with other islands.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by PedroAugustoRibas — APPROVED** [review]

File: packages/nuxt/src/app/components/nuxt-island.ts [hunk 1]
```diff
@@ -64,6 +64,14 @@ export default defineComponent({
       type: String,
       default: () => undefined
     },
+    /**
+     * use the NuxtIslandResponse which has been cached if available
+     * @default true
+     */
+    useCache: {
```
Comment (manniL) [inline packages/nuxt/src/app/components/nuxt-island.ts:71]: Maybe just me but "useCache" sounds like a composable on the first read :D

**Comment by huang-julien** [comment]
After some discussions, this would probably not be used by many people so i'm going to close it. We probably will re-open it once people ask about this feature

**Comment by Rigo-m** [comment]
Is it possible to cache islands using routeRules instead?

**Comment by huang-julien** [comment]
This is only for  NuxtIsland to always fetch the  island component instead of reusing a previous result

**Pull request closed without merging**